### DISCUSSION TITLE
Change default behavior for Modal Popup

### DISCUSSION THREAD

**Discussion by Sharafudeen** [oqtane/oqtane.framework#3332] [General]
I’ve implemented a datagrid with edit action link. If I click edit link, the datagrid is disappeared and shows modal popup dialog. How do I change this default behavior for Modal Popup. Is there any way to show modal popup without disappearing the datagrid?

**Comment by nicpitsch** [comment]
I have been faced with a similar issue (based on your description) using Telerik Grid. The helper popup's in edit mode, e.g. for date input, aren't functional. Edit components in Oqtane are rendered based on [Bootstrap Modal](https://getbootstrap.com/docs/5.3/components/modal/) (ActionDialog.razor component in Oqtane). I propose to try to set `UseAdminContainer` property in your edit component to false to see if that's the right direction concerning your issue. Look for *UseAdminContainer* within the discussiuns here. 

Setting `UseAdminContainer` to false was not a solution for me. As a (temporary) workaround for Telerik Grid I'm using for such columns an `EditorTemplate` with Blazor `InputDate` component instead.

Which *datagrid* component do you use? In my understanding it's not a Oqtane issue, but an issue about how the third party components behave within Bootstrap Modal.

During searching a solution I found somewhere a discussion concerning that Bootstrap has changend the *Modal* behaviour in one of the recently versions, setting it more dominant (I don't find it anymore).

**Reply by Sharafudeen** [reply]
I’ve tried two cases as follows,

1) Using HTML table which is the default datagrid in Oqtane without changing anything that’s also having same disappearing problem.

2) Using Havit.Blazor UI component that’s also having same problem.

What I guess (I’m not sure), this issue maybe because of Oqtane’s dynamic routing? I’ve notified this in the browser address bar when clicking the “edit link” which is creating new routes.

**Comment by sbwalker** [comment]
@Sharafudeen if you search for UseAdminContainer in the Discussions you will find solutions which enable you to avoid using modal dialogs (the default behavior in Oqtane).
